GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -229,6 +229,18 @@ public virtual void Shutdown()
             this.indexParsingThread.Wait();
         }
 
+        public void AllowIndexToBeParsed()

```

Comment:
This is only not async because the code has a `Wait`.  It is allowing the index parsing thread to run and parse the index.  What would be a better name? `AllowIndexParsingThreadToRun`,  `ReleaseIndexParsingThreadWait`?

Reply:
Maybe `WakeIndexParsingThreadAndWait`?  I like your suggestion of `ReleaseIndexParsingThreadWait` as well.